src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -960,14 +960,16 @@ class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
 
       val policyRole = PolicyRoleTable.syntax("policyRole")
       val resourceRole = ResourceRoleTable.syntax("resourceRole")
+      val flattenedRole = FlattenedRoleMaterializedView.syntax("flattenedRole")
       val policyActionJoin = PolicyActionTable.syntax("policyActionJoin")
       val policyAction = ResourceActionTable.syntax("policyAction")
 
       val listUserResourcesQuery = samsql"""$cteQueryFragment
         select ${userResourcePolicy.result.baseResourceName}, ${resourceRole.result.role}, ${policyAction.result.action}, ${userResourcePolicy.result.public}, ${userResourcePolicy.result.inherited}
           from ${userResourcePolicyTable as userResourcePolicy}
-          left join ${PolicyRoleTable as policyRole} on ${userResourcePolicy.policyId} = ${policyRole.resourcePolicyId} and ${userResourcePolicy.inherited} = ${policyRole.descendantsOnly}
-          left join ${ResourceRoleTable as resourceRole} on ${policyRole.resourceRoleId} = ${resourceRole.id} and ${userResourcePolicy.baseResourceTypeId} = ${resourceRole.resourceTypeId}
+          left join ${PolicyRoleTable as policyRole} on ${userResourcePolicy.policyId} = ${policyRole.resourcePolicyId}
+          left join ${FlattenedRoleMaterializedView as flattenedRole} on ${policyRole.resourceRoleId} = ${flattenedRole.baseRoleId} and ((${userResourcePolicy.inherited} and (${policyRole.descendantsOnly} or ${flattenedRole.descendantsOnly})) or not (${userResourcePolicy.inherited} or ${policyRole.descendantsOnly} or ${flattenedRole.descendantsOnly}))
```

Comment:
I don't really care for this long logic statement, but I believe it is _right_. Would love to shorten/simplify it if anyone has any ideas

Reply:
well I tried this and it did not quite work so I am missing something

Reply:
I had to draw out a truth table because I kept making minor changes that would fix one test and break another.
<img width="512" alt="Screen Shot 2020-11-13 at 9 24 56 AM" src="https://user-images.githubusercontent.com/19961550/99082438-27e98380-2592-11eb-8df8-b598c90d1e61.png">

Reply:
I couldn't move everything into the where clause because it's just on a left join here and the where clause is too selective, but I did move it into a where clause for `listUserResourceActions` and `listUserResourceRoles`

Reply:
Yeah, I think it would be helpful. I'm pulling this check out into a separate query fragment and adding a comment